### DISCUSSION TITLE
FastAPI performance testing with jmeter issue regarding params and request body

### DISCUSSION THREAD

**Discussion by akshats2851** [fastapi/fastapi#9632] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Not required
```

### Description

I am trying to do performance testing using jmeter using parameters. FastAPI throws error showing the body is missing. I don't want to send request body but fastapi does not allow it. Is there any way to disable request body or send an empty object in jmeter along with parameters. Also, jmeter only allows sending either parameters or request body.
Error: 
{
    "detail": [
        {
            "loc": [
                "body"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}

### Operating System

Windows

### FastAPI Version

0.95.2

### Python Version

3.7.3

**Comment by jgould22** [comment]
Perhaps we can help more with some clarification.

You are attempting to load test an application with jmeter or just a specific endpoint?

What kind of end point are you trying to load test GET or POST, both?

Does your endpoint require things in the body?

A code sample would be helpful.

**Comment by akshats2851** [comment]
So, i am testing multiple apis, all GET using jmeter. The endpoint does not require body params, only query params are needed.
Even for hitting an api using postman, i need to send an empty object {} as body along with params, otherwise it shows the same error as mentioned above.
code sample:
app = FastAPI()
@app.get("endpoint")
async def main_func(input: dict, request: Request):
       ----body of function----
       return value

**Comment by YuriiMotov** [comment, marked as answer]
> code sample:
>  ```
> app = FastAPI()
> @app.get("endpoint")
> async def main_func(input: dict, request: Request):
>     # ----body of function----
>     return value

`dict` is complex type and it's treated by FastAPI as `Body` parameter
